### PR TITLE
added zendesk_app to .gitignore

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description

New method on deploy model used by zendesk_app

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low/Med/High

added 2 new methods to deploy model

### References
* JIRA: 

### Risks
* [HIGH | medium | low] Does it work across browsers (including IE!)?
* [HIGH | medium | low] Does it work in the different products (Lotus, Zopim, People)?
* [HIGH | medium | low] Are there any performance implications?
* [HIGH | medium | low] Any security risks?
* [HIGH | medium | low] What features does this touch?

### CODE REVIEW

File: app/models/deploy.rb [hunk 1]
```diff
@@ -97,6 +97,16 @@ def pending_start!
     DeployService.new(user).confirm_deploy!(self)
   end
 
+  def self.in_range(time_value, time_unit)
```
Comment (grosser) [inline app/models/deploy.rb]: add tests if you add code ... or move the code into the plugin ?

File: .gitignore [hunk 2]
```diff
@@ -37,5 +37,8 @@
 # Ignore temp devel files
 /routes.txt
 
+# Ignore certain plugins
```
Comment (grosser) [inline .gitignore]: might be a good idea to ignore all plugins to make experimentation easier ...

Comment (grosser) [inline .gitignore]: alternatively ignore all /_* folders ... then we can link them to a _ name ...

File: app/models/deploy.rb [hunk 3]
```diff
@@ -97,6 +97,16 @@ def pending_start!
...
+    where(created_at: (Time.now - time_value.public_send(time_unit))..Time.now)
```
Comment (grosser) [inline app/models/deploy.rb]: .ago instead of time math

Comment (grosser) [inline app/models/deploy.rb]: also can be simplified to the faster `"created_at > ?", xxx.ago` (1 time comparison vs 2)

File: app/models/deploy.rb [hunk 4]
```diff
@@ -97,6 +97,16 @@ def pending_start!
...
+  end
+
+  def self.search_by_project(search)
+    return self unless search
+    joins({stage: 'project'})
+    .where('lower(projects.name) like ?', "%#{sanitize_sql_like(search.downcase)}%")
```
Comment (grosser) [inline app/models/deploy.rb]: indent on continuation

File: app/models/deploy.rb [hunk 5]
```diff
@@ -97,6 +97,16 @@ def pending_start!
     DeployService.new(user).confirm_deploy!(self)
   end
 
+  def self.in_range(time_value, time_unit)
+    where(created_at: (Time.now - time_value.public_send(time_unit))..Time.now)
+  end
+
+  def self.search_by_project(search)
+    return self unless search
+    joins({stage: 'project'})
```
Comment (grosser) [inline app/models/deploy.rb]: no need for `{}` for option hashes

File: app/models/deploy.rb (hunk 4 above)
Comment (grosser) [inline app/models/deploy.rb]: posibly faster / cleaner to search all the matching projects + `.pluck(:id)` and then select stages

**Comment by sandlerr** [comment]
I'd suggest keeping the `Deploy` scopes in your plugin, that makes it less likely that someone deletes them thinking they are unused code.
[Here](https://github.com/zendesk/samson/blob/master/plugins/env/app/decorators/project_decorator.rb)'s an example of patching core classes in a plugin.

**Pull request closed without merging**